src/coreclr/pal/src/thread/process.cpp

```diff
@@ -2575,20 +2605,11 @@ BOOL
 InitializeFlushProcessWriteBuffers()
 {
     _ASSERTE(s_helperPage == 0);
-    _ASSERTE(s_flushUsingMemBarrier == 0);
 
-    // Starting with Linux kernel 4.14, process memory barriers can be generated
-    // using MEMBARRIER_CMD_PRIVATE_EXPEDITED.
-    int mask = membarrier(MEMBARRIER_CMD_QUERY, 0);
-    if (mask >= 0 &&
-        mask & MEMBARRIER_CMD_PRIVATE_EXPEDITED)
+    InitializeMembarrier();
```

Comment:
Instead of splitting the initialization into two methods, could you please move the call to the existing `InitializeFlushProcessWriteBuffers` method to somewhere around here: https://github.com/dotnet/runtime/pull/106724/files#diff-ed6f693a713e751b280574475591c567e6adfbdb97b02a814411c0de8b9ae2e0R388

```
if (flags & PAL_INITIALIZE_FLUSH_PROCES_WRITE_BUFFERS)
{
    if (!InitializeFlushProcessWriteBuffers())
    {
        palError = ERROR_PALINIT_INITIALIZE_FLUSH_PROCESS_WRITE_BUFFERS;
        goto CLEANUP0a;
    }
}
```

+ define `PAL_INITIALIZE_FLUSH_PROCES_WRITE_BUFFERS` in pal.h:

`#define PAL_INITIALIZE_REGISTER_ACTIVATION_SIGNAL   0x100` and add it to `PAL_INITIALIZE_CORECLR` set of flags in the same file.

Reply:
Pushed a new rev with your suggestion